simulation/src/test/java/com/palantir/dialogue/core/Strategy.java

```diff
@@ -0,0 +1,133 @@
+            List<LimitedChannel> limitedChannels1 = channels.stream()
+                    .map(c1 -> new ConcurrencyLimitedChannel(
+                            c1, () -> ConcurrencyLimitedChannel.createLimiter(sim.clock()::read)))
+                    .collect(Collectors.toList());
+            LimitedChannel limited1 = new RoundRobinChannel(limitedChannels1);
+            limited1 = instrumentClient(limited1, sim.metrics()); // just for debugging
+            Channel channel = new QueuedChannel(limited1, DispatcherMetrics.of(new DefaultTaggedMetricRegistry()));
+            return new RetryingChannel(channel);
+        });
+    }
+
+    private static Channel roundRobin(Simulation sim, Supplier<List<SimulationServer>> channelSupplier) {
+        return RefreshingChannelFactory.RefreshingChannel.create(channelSupplier, channels -> {
+            List<LimitedChannel> limitedChannels =
+                    channels.stream().map(Strategy::noOpLimitedChannel).collect(Collectors.toList());
+            LimitedChannel limited = new RoundRobinChannel(limitedChannels);
+            limited = instrumentClient(limited, sim.metrics()); // will always be zero due to the noOpLimitedChannel
+            Channel channel = new QueuedChannel(limited, DispatcherMetrics.of(new DefaultTaggedMetricRegistry()));
+            return new RetryingChannel(channel);
+        });
+    }
+
+    private static LimitedChannel instrumentClient(LimitedChannel delegate, SimulationMetrics metrics) {
+        Meter starts = metrics.meter("test_client.starts");
+        Counter metric = metrics.counter("test_client.refusals");
+        return new LimitedChannel() {
+
+            @Override
+            public Optional<ListenableFuture<Response>> maybeExecute(Endpoint endpoint, Request request) {
+                log.info("starting traceid={}", request.headerParams().get("X-B3-TraceId"));
+                starts.mark();
+                Optional<ListenableFuture<Response>> response = delegate.maybeExecute(endpoint, request);
+                if (!response.isPresent()) {
+                    metric.inc();
+                }
+                return response;
+            }
+
+            @Override
+            public String toString() {
+                return delegate.toString();
+            }
+        };
+    }
+
+    /** This is an alternative to the {@link com.palantir.dialogue.core.QueuedChannel}. */
+    static Channel dontTolerateLimits(LimitedChannel limitedChannel) {
+        return new Channel() {
+            @Override
+            public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+                Optional<ListenableFuture<Response>> future = limitedChannel.maybeExecute(endpoint, request);
+                if (future.isPresent()) {
+                    return future.get();
+                }
+
+                return Futures.immediateFailedFuture(new SafeRuntimeException("limited channel says no :("));
```

Comment:
I think queued channel should do this #351

Reply:
Yeah I was using this as a substitute for the queued channel, but it's not being used right now, so will delete and bring it back when needed.